feat: enable verification of certs via `network.tls_verify` and priva…

schakko opened the pull request:
feat: enable verification of certs via `network.tls_verify` and privte CA chains with `network.tls_ca_path`

This patch allows us 
- to use `network.tls_verify = true|false` or enable or disable x509 certificate verfication
- setting `network.tls_ca_path  <path>` to a custom CA bundle when using a private Certificate Authority.

christiaangoossens commented on custom_components/auth_oidc/__init__.py:
```diff
@@ -50,17 +51,19 @@ async def async_setup(hass: HomeAssistant, config):
     _LOGGER.info("Registered OIDC provider")
 
-    # We only use openid, profile & groups, never email
-    scope = "openid profile groups"
```
You probably based your PR of the wrong branch, can you rebase?

christiaangoossens replied:
Seems like this change is still there, any reason why you need this scope removed (maybe it doesn't work with Keycloak)?

If you need that, we should make a config flag to set the scopes instead which defaults to openid email groups.

christiaangoossens replied:
Comment still changed.

christiaangoossens reviewed (changes requested):
Rebase from main seems necessary

christiaangoossens commented:
@schakko I am not against this change, but I would like some more docs in the README. What is this useful for? Maybe running an OIDC provider locally without public TLS certs? 

Also warn users against using tls_verify false, especially when setting a CA path is possible.

schakko commented:
@christiaangoossens Sure, I
- rebased on your latest changes
- added an example in the README.md with a hint for `network.tls_verify: false`
- also set the default value of `network.tls_verify` to `true` if that configuration setting is not present.

Let me know if you need any other changes!

christiaangoossens commented:
One comment left in the code and a failing lint pipeline

christiaangoossens commented:
@schakko Pipeline says that you should run `rye run fix` first to format your files and then check it again with `rye run check`.

christiaangoossens commented:
Merged! If you make another PR, no need to force push, I will squash them anyway ;)
Thank you for your contribution!
